modules/spring-config/src/test/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBeanTestCase.java

```diff
@@ -250,95 +272,182 @@ private Processor getFlowRefProcessor(FlowRefFactoryBean factoryBean) throws Exc
 
   @Test
   public void dynamicFlowRefDoesNotExist() throws Exception {
-    doReturn(true).when(expressionManager).isExpression(anyString());
     doReturn(new TypedValue<>("other", STRING)).when(expressionManager).evaluate(eq(DYNAMIC_NON_EXISTANT), eq(DataType.STRING),
-                                                                                 eq(NULL_BINDING_CONTEXT), any(CoreEvent.class),
-                                                                                 any(ComponentLocation.class), eq(true));
+            eq(NULL_BINDING_CONTEXT), any(CoreEvent.class),
+            any(ComponentLocation.class), eq(true));
 
     expectedException.expect(instanceOf(RoutePathNotFoundException.class));
-    getFlowRefProcessor(createFlowRefFactoryBean(DYNAMIC_NON_EXISTANT)).process(testEvent());
+    getFlowRefProcessor(createFlowRefFactoryBean(DYNAMIC_NON_EXISTANT, "flow", applicationContext)).process(testEvent());
   }
 
-  private FlowRefFactoryBean createFlowRefFactoryBean(String name) throws Exception {
+  @Test
+  public void concurrentDynamicSubFlowInstantiation() throws Exception {
```

Comment:
add allure `@Issue`

Reply:
Added new annotations to AllureConstants